webapp/tests/karma/ts/modules/analytics/analytics.component.spec.ts

```diff
@@ -0,0 +1,125 @@
+    };
+    analyticsActions = {
+      setSelectedAnalytics: sinon.stub(AnalyticsActions.prototype, 'setSelectedAnalytics'),
+      setAnalyticsModules: sinon.stub(AnalyticsActions.prototype, 'setAnalyticsModules')
+    };
+    activatedRoute = {
+      snapshot: {
+        routeConfig: { path: 'targets' },
+      },
+    };
+
+    TestBed.configureTestingModule({
+      declarations: [ AnalyticsComponent ],
+      imports: [
+        RouterTestingModule,
+      ],
+      providers: [
+        provideMockStore({ selectors: mockSelectors }),
+        { provide: AnalyticsModulesService, useValue: analyticsModulesService },
+        { provide: ActivatedRoute, useValue: activatedRoute },
+      ]
+    })
+    .compileComponents()
+    .then(() => {
+      fixture = TestBed.createComponent(AnalyticsComponent);
+      component = fixture.componentInstance;
+      analyticsModulesService = TestBed.inject(AnalyticsModulesService);
+      router = TestBed.inject(Router);
+      fixture.detectChanges();
+    });
+  }));
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  it('should create and set up controller with no modules', () => {
+    expect(component).to.exist;
+    expect(component.analyticsModules).to.be.empty;
+    expect(analyticsActions.setSelectedAnalytics.callCount).to.equal(2);
+    expect(globalActions.unsetSelected.callCount).to.equal(1);
+    expect(analyticsModulesService.get.callCount).to.equal(1);
+  });
+
+  it('should set selected the specified module', async () => {
+    sinon.reset();
+    const analyticsModules = [
+      { route: 'reporting' },
+      { route: 'targets' }
+    ];
+    analyticsModulesService.get.resolves([...analyticsModules]);
+
+    await component.getAnalyticsModules();
+
+    expect(analyticsModulesService.get.callCount).to.equal(1);
+    expect(analyticsActions.setSelectedAnalytics.callCount).to.equal(1);
+    expect(analyticsActions.setSelectedAnalytics.args[0][0]).to.equal(analyticsModules[1]);
+    expect(analyticsActions.setAnalyticsModules.callCount).to.equal(1);
+    expect(analyticsActions.setAnalyticsModules.args[0][0]).to.include.members(analyticsModules);
+  });
+
+  it('should jump to child route if single module is present', async () => {
+    sinon.reset();
+    activatedRoute.snapshot.routeConfig.path = 'analytics';
+    const navigateStub = sinon.stub(router, 'navigate');
+    const analyticsModules = [{ route: 'targets' }];
+    analyticsModulesService.get.resolves(analyticsModules);
+
+    await component.getAnalyticsModules();
+
+    expect(analyticsModulesService.get.callCount).to.equal(1);
+    expect(analyticsActions.setSelectedAnalytics.callCount).to.equal(0);
+    expect(analyticsActions.setAnalyticsModules.callCount).to.equal(1);
+    expect(analyticsActions.setAnalyticsModules.args[0][0]).to.include.members(analyticsModules);
+    expect(navigateStub.callCount).to.equal(1);
+    expect(navigateStub.args[0][0][0]).to.equal('targets');
```

Comment:
Can we please make this assertion more specific (assert arguments in the entirety)? The entire path matters where it comes to navigation, so I think it's important to test if we navigate to ['targets'] or to ['targets', 'thing']. 
Like `expect(navigateStub.args[0].to.deep.equal([['targets']]))`;

Reply:
Oh yes, fixed! Thanks